### ISSUE TITLE
Example of Full Waveform Inversion

### ISSUE THREAD

**Issue by mlin26** [geodynamics/specfem3d#1380]
Hi,

I read through your manual and it says you have FWI examples in the EXAMPLES folder. But I'm not able to find it. Could you tell me where the FWI examples are? Thanks!

**Comment by danielpeter** [comment]
yes, sorry i'm afraid that has never been finalized. let me work out a small example and provide it soon... 

in the past, the SPECFEM codes have mostly focussed on computing forward/adjoint simulations, less so on solving an iterative, non-linear optimization problem. there has been work to include an inversion code, but it requires at the moment to look up the source code in order to understand how it would need to be setup and done. let me do that and see if a simple example could help.

**Comment by mlin26** [comment]
Thank you! I'd appreciate it if you could provide both 2d and 3d example.
